app/javascript/components/HumanOverrideChatWindow.jsx

```diff
@@ -146,21 +163,41 @@ class InputMessage extends Component {
   }
 }
 
-const ChatWindowComponent = ({ sendMessage, bot, message, inputRef }) => (
-  <Paper
+const ChatWindowComponent = ({ sendMessage, resolveMessage, bot, message, inputRef }) => {
+
+  const messages = () => {
+
+    const textMessages = (messages) => (
+      messages.filter(
+        msg => msg.type == 'text'
+      ).sort(
+        (a, b) => a.timestamp < b.timestamp ? -1 : 1
+      ).map(
+        msg => ({ text: msg.content, sent: msg.direction == 'otu' })
+      )
+    )
+
+    return [
+      ...[{text: message.data.message, sent: false}],
+      ...message.data.messages ? textMessages(message.data.messages) : []
+    ]
+  }
+
+  return <Paper
     zDepth={5}
     className={"chat-window"}>
       <ChatHeader title={message.data.name || 'Unknown'} subtitle={bot.name} />
-      <MessageList messages={[{id: message.id, text: message.data.message, sent: false, timestamp: message.created_at}]} />
-      <InputMessage onSend={sendMessage} ref={inputRef}/>
+      <MessageList messages={messages()} />
+      <InputMessage onSend={sendMessage} onResolve={resolveMessage} ref={inputRef}/>
```

Comment:
The resolve button should be on the ChatHeader according to the mockups, but we can move it later.